pkg/kademlia/kademlia_test.go

```diff
@@ -187,7 +200,11 @@ func testNode(t *testing.T, bn []pb.Node) (*Kademlia, *grpc.Server) {
 
 	pb.RegisterNodesServer(grpcServer, s)
 	go func() { _ = grpcServer.Serve(lis) }()
-
 	return k, grpcServer
 
 }
+
+func cleanup(t *testing.T, rt *RoutingTable, dir string) {
+	assert.NoError(t, rt.Close())
```

Comment:
This should happen in Kademlia code, not just in test-code. If we do not close things properly in production we still leak things.

Reply:
will do in another pr https://storjlabs.atlassian.net/browse/V3-528

Reply:
We shouldn't be hiding problems in tests. Fixing Kademlia disconnect is part of the fix to the tests.

Reply:
The original scope of this PR was to remove files & directories from being left behind from tests. 

I agree that we should be properly handling disconnections and avoiding leaks but that should be another ticket and not jammed into this one.

Reply:
It was about ensuring that tests don't leave garbage behind. I think I might have not been clear about the purpose of not having garbage.

Yes one part of it is not to commit accidentally test data when the test fails. But the more important goal of tests not leaving garbage around is not really about the tests, it's more about ensuring that our production system can properly clean up after itself. Not cleaning up after ourselves has already caused problems with the piecestore intermittently failing.

I'm fine with handling the disconnecting in a separate PR. But currently it doesn't also cleanup and shutdown the databases.

Reply:
Completely agree that we should do this. I just don't want this PR to grow too large.

Reply:
👍